packages/core/src/App/Containers/new-version-notification.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+/**
+ * Listens on emitted events to show notification on available updates.
+ *  - IgnorePWAUpdate event will prevent notification
+ *  - ListenPWAUpdate event is usefull to re-assign notify on update behavior
+ * @param onUpdate
+ * @return {null}
+ * @constructor
+ */
+const NewVersionNotification = ({ onUpdate }) => {
+    React.useEffect(() => {
+        const removeUpdateListener = () => {
+            document.removeEventListener('UpdateAvailable', onUpdate);
+        };
+
+        const addUpdateListener = () => {
+            document.addEventListener('UpdateAvailable', onUpdate);
+        };
+
+        document.addEventListener('IgnorePWAUpdate', removeUpdateListener);
+        document.addEventListener('ListenPWAUpdate', addUpdateListener);
+
+        return () => {
+            document.removeEventListener('IgnorePWAUpdate', removeUpdateListener);
+            document.removeEventListener('ListenPWAUpdate', addUpdateListener);
+        };
+    });
+
+    React.useEffect(() => {
```

Comment:
- Couldn't both of these hooks be consolidated into a single one? 
- Should we define these effects to only run on a mount, and not on each re-render? Or will this child not re-render because the props aren't changing?

Reply:
Both are great issues. thanks for the feedback. I've merged them into a single commit.